actioncheck/example_test.go

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package actioncheck_test
+
+import (
+	"github.com/hashicorp/terraform-plugin-testing/actioncheck"
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+)
+
+// Example of how to use action checks in a test case
+func ExampleExpectProgressMessageContains() {
+	// This is how you would use action checks in a real test
+	_ = resource.TestCase{
+		Steps: []resource.TestStep{
+			{
+				Config: `
+					action "aws_lambda_invoke" "test" {
+						config {
+							function_name = "my-function"
+							payload       = "{\"key\":\"value\"}"
+							log_type      = "Tail"
+						}
+					}
+				`,
+				ActionChecks: []actioncheck.ActionCheck{
+					// Check that the action produces a message containing log output
+					resource.TestCheckProgressMessageContains("aws_lambda_invoke.test", "Lambda function logs:"),
```

Comment:
```suggestion
					resource.TestCheckProgressMessageContains("aws_lambda_invoke", "Lambda function logs:"),
```

The examples as written do not work because the provider does not have access to the alias name so you can only reference messages by action type name.